restore/data.go

```diff
@@ -227,8 +230,15 @@ func restoreDataFromTimestamp(fpInfo filepath.FilePathInfo, dataEntries []toc.Co
 
 			setGUCsForConnection(gucStatements, whichConn)
 			for entry := range tasks {
+				// Check if any error occurred in any other goroutines:
+				select {
+				case <-ctx.Done():
+					return // Error somewhere, terminate
```

Comment:
Why did not you add `cancel()` here?

Reply:
because cancels in other places lead to this case of select